### ISSUE TITLE
"ui_accept" InputEvent not working when code generated

### ISSUE THREAD

**Issue by Krzycho666** [okamstudio/godot#1941]
I'm trying to push some custom action events to use some default buttons behaviour. Since there was no problem about "ui_left/right/up/down/escape", I cant manage to make "ui_accept" work. This is the code I use:

```
var event = InputEvent()
event.type = InputEvent.ACTION
event.set_as_action( "ui_accept", true )
get_tree().input_event( event )
```
It seems to me that there is some bug that I cant find. Corrent me if I'm wrong.

**Comment by reduz** [comment]
do you have a control focused? ui_accept will do nothing if no control has input focus

**Comment by okamstudio** [comment]
try doing:

some_control.grab_focus()

and then send the ui_accept

On Tue, May 19, 2015 at 9:11 PM, Juan Linietsky <notifications@github.com>
wrote:

> do you have a control focused? ui_accept will do nothing if no control has
> input focus
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/okamstudio/godot/issues/1941#issuecomment-103702426>.
>



-- 

*OkamStudio*

**Comment by adolson** [comment]
Is there something special about ui_accept?

**Comment by Krzycho666** [comment]
There is always one button with grabbed focus on it in my scene. It starts with already grabbed focus on the first button - I do it manualy in the code. I tried even clicking on the button, and releasing it outside to grab the focus, but no change. As I mentioned, there is no problem with other ui actions, like ui_left/right/up/down/escape, so I can already freely nivigate my menu through. I've even tried to look into button class source - but I cant see why does it happen. Does my code work for you?

**Comment by Krzycho666** [comment]
Well... I've found the bug - but in Godot itself.
In BaseButton implementetion, in _input_event method, there is a switch(p_event.type). The problem is, that current implementation accepts ui_accept only when event is KEY typed - not ACTION. It's quite strange, cose there is a this check inside:
```
if (p_event.is_action("ui_accept")) 
```
while in that switch case there is a chack if event is KEY typed, so it's a paradox. After changing this case to ACTION it works fine, but the default mappings stopped to work, because of how are they done ( they are KEY actions this action added - it shouldn't be done that way. Should it? )
It quite messy out there about this. Netherless I've found what it was all about. It just can't work.
